### ISSUE TITLE
Trouble with double puppeting - sent messages not appearing on bridge

### ISSUE THREAD

**Issue by ajkessel** [duo/matrix-wechat#19]
I can't get messages sent from the WeChat app to show up on Matrix. I've tried both shared secret authentication as well as the login-matrix method. Other bridges (e.g. WhatsApp, iMessages) are working fine for double puppeting and posting sent messages, so I'm not sure what's difference with the WeChat bridge.

I see this in my log:

```
[Aug 16, 2023 02:41:26] [Puppet/wxid_k5meyls4mino12u/DEBUG] Syncing info through wxid_k5meyls4mino12u
[Aug 16, 2023 02:41:26] [Portal/wxid_k5meyls4mino12u/DEBUG] Not handling 4487028428340278489: user doesn't have double puppeting enabled
```

**Comment by duo** [comment]
If you are using the shared secret method, please check if the configurations for `double_puppet_server_map` and `login_shared_secret_map` in the bridge are correct.

**Comment by ajkessel** [comment]
I believe they are correct. I have the following in the wechat `config.yaml`:
```
    double_puppet_server_map:
        mydomain.net: https://matrix.mydomain.net
```
(My Matrix username is in the form user@mydomain.net and the domain name for the synapse homeserver is matrix.mydomain.net)
```
    login_shared_secret_map:
        mydomain.net: [shared secret key that is working for other bridges]
```
And in `homeserver.yaml` I have:
```
modules:
   - module: shared_secret_authenticator.SharedSecretAuthProvider
     config:
       shared_secret: "[same key used above]"
       m_login_password_support_enabled: true
```
Am I doing something wrong?

**Comment by ajkessel** [comment]
Update -- I think I figured it out. It does seem to be working generally now that I have the shared secret configured as above. I was using a room that was a DM to myself for testing purposes and those messages aren't getting puppeted, but messages with DMs to other people are working fine, so it must be some artifact of how the bridge or WeChat handles messages to oneself.

**Comment by duo** [comment]
Huh, I don't even know how WeChat client can send messages to oneself...

**Comment by ajkessel** [comment]
Yes, you can just select yourself in WeChat contacts. But in any case I think this issue can be closed.
